### DISCUSSION TITLE
Variable width fonts

### DISCUSSION THREAD

**Discussion by Etrusan** [Chlumsky/msdf-atlas-gen#32] [Q&A] [answered]
How would I specify which typeface to use when using variable width fonts? Microsoft Bahnschrift is a good example.

**Comment by Chlumsky** [comment]
Fonts are handled by FreeType. I don't know if they support variable fonts and to what extent, but this project was not made with variable fonts in mind at all. If it's just a matter of passing some variables along to FreeType, support can be added easily, but otherwise most likely not.

**Comment by Etrusan** [comment]
All typefaces are packed in a single .ttf. I don't understand fonts so I can't tell you more unfortunately.

//[According to wiki, FreeType received Variable font support in 2.8](https://en.wikipedia.org/wiki/Variable_font)

**Comment by Etrusan** [comment]
Heyo, any update on this?

**Reply by Chlumsky** [reply]
No. You can either add support for it yourself - [here are the fonts loaded](https://github.com/Chlumsky/msdfgen/blob/64a91eec3ca3787e6f78b4c99fcd3052ad3e37c0/ext/import-font.cpp#L117-L141) and [here is FreeType documentation](https://freetype.org/freetype2/docs/reference/ft2-multiple_masters.html) (you may need to use a newer version of FreeType than what's in the repo), or use a [font converter](https://fonttools.readthedocs.io/en/latest/varLib/instancer.html) where you can specify the font variables and generate a regular font and feed that into the generator.

**Comment by Chlumsky** [comment, marked as answer]
I know this is over a year old thread, but thanks to a user's pull request to MDSFgen, I ported this functionality to the atlas generator today and it is now available.
